realm/realm-library/src/objectServer/java/io/realm/internal/SyncObjectServerFacade.java

```diff
@@ -91,7 +91,7 @@ public void realmClosed(RealmConfiguration configuration) {
             // make sure the user is still valid
             SyncUser user = syncConfig.getUser();
             if (!user.isValid()) {
-                if (user.getAccessToken() == null) {
+                if (user.isLoggedOut()) {
```

Comment:
I think this is the wrong change. Users should be to open a Realm even if not logged in. Otherwise, they might not be able to open the Realm if offline.